### DISCUSSION TITLE
"Push Notifications Guide" doesn't mention AppDelegate.swift

### DISCUSSION THREAD

**Discussion by adhalejr** [capawesome-team/capacitor-firebase#394] [Q&A]
This article is quite helpful (though I'd love a React version): https://capawesome.io/blog/the-push-notifications-guide-for-capacitor/

But it doesn't say anything about required changes to `AppDelegate.swift`.

The documentation for the plugin itself: https://capawesome.io/plugins/firebase/cloud-messaging/#ios

mentions multiple changes to that file, along with possible / optional changes to that file and to `Info.plist`.

Which is correct? Will things fail completely if `AppDelegate.swift` doesn't have those changes?

**Comment by robingenz** [comment]
Thank you for pointing this out. I have updated the blog post. This was indeed missing.

**Reply by adhalejr** [reply]
I appreciate the reply. A github repo with **this particular article's specific code** would be terrifically useful. The demo app linked from the plugin's github page doesn't match this article.

Thanks again.

**Reply by robingenz** [reply]
> The demo app linked from the plugin's github page doesn't match this article.

What makes you think that? I created the demo app along with the article and everything should match.

**Reply by adhalejr** [reply]
The guide has a link to the plugin: https://capawesome.io/plugins/firebase/cloud-messaging/

That page has a link to a demo / "working example": https://github.com/robingenz/capacitor-firebase-plugin-demo

When I look at that demo as compared to the guide, I don't see:

- A `google-services.json` file under the `android` project / folder
- A `GoogleService-Info.plist` file under the `ios` project / folder
- The `src/app/homehome.page.ts` file referenced in the article
- etc.

If I'm misunderstanding something, please let me know.

IMO, it would be helpful if the guide article itself had a link to the demo / working example that exactly matched the final result of what the article documents.

Thank you again.

**Reply by robingenz** [reply]
The guide also has a link to this demo app: https://github.com/robingenz/ionic-capacitor-firebase-messaging-demo

> Thanks to the [Capacitor Firebase Cloud Messaging](/plugins/firebase/cloud-messaging/) plugin, once successfully configured, you get a single API for all three platforms.
You can find the complete app on [GitHub](https://github.com/robingenz/ionic-capacitor-firebase-messaging-demo).

**Reply by adhalejr** [reply]
Thank you. I looked for such a link in the guide but didn't find it. I've found it now, and that demo app seems to have everything from the article.

Thanks again.
